### PR TITLE
LEARN-23542 Homepage

### PR DESCRIPTION
**Pull request by abhijit-deb-kaplan**
JIRA : 
https://kna-core.atlassian.net/browse/LEARN-23542
https://kna-core.atlassian.net/browse/LEARN-23543
https://kna-core.atlassian.net/browse/LEARN-23544
https://kna-core.atlassian.net/browse/LEARN-23545

# Summary
- Added Homepage functionalities UI
- Updated to BEM conventions
- Added signup functionalities
- Added POST api to user data to database
- Created services and constants for API and URLs
- Added login functionalities 
- Fetched user details on successful login
- Created services for API calls

### CODE REVIEW

File: src/app/login-dialog/login-dialog.component.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  &__heading {
+    display: flex;
+    width: 100%;
+    justify-content: space-between;
+    align-items: center;
+    font-size: 25px;
+    margin-bottom: 20px;
+
+    & p {
```
Comment (MuskaaanBansal) [inline src/app/login-dialog/login-dialog.component.scss]: don't use tags to define property. Here p tag is directly being used, create a class for the same, please make this change in all css files, there should't be any tags being directly used in any css files, like: p,img h1.. h7, a, buttons etc

**Comment by MuskaaanBansal** [comment]
Screen Shot 2023-10-03 at 11 16 35 AM
Add cursor pointer to the buttons, right now when I hover on the buttons, I see an arrow

**Comment by MuskaaanBansal** [comment]
I am able to click on sign up and login buttons even without entering the required fields, you need to disable the buttons when the form is invalid

File: src/app/login-dialog/login-dialog.component.scss (hunk 1 above)
Comment (abhijit-deb-kaplan) [reply src/app/login-dialog/login-dialog.component.scss]: Fixed that

File: src/app/components/login-dialog/login-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import {MatProgressSpinnerModule} from '@angular/material/progress-spinner';
+import { HttpClient } from '@angular/common/http';
+import { UserApiService } from '../../services/users/user-api.service';
+
+@Component({
+  selector: 'app-login-dialog',
+  templateUrl: './login-dialog.component.html',
+  styleUrls: ['./login-dialog.component.scss'],
+  standalone: true,
+  imports: [MatFormFieldModule, MatInputModule, FormsModule, ReactiveFormsModule, NgIf, MatIconModule, MatButtonModule, MatSnackBarModule,MatProgressSpinnerModule],
```
Comment (MuskaaanBansal) [inline src/app/components/login-dialog/login-dialog.component.ts]:
```suggestion

```

Comment (MuskaaanBansal) [reply src/app/components/login-dialog/login-dialog.component.ts]: add them to app module

File: src/app/components/login-dialog/login-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+})
+export class LoginDialogComponent {
+  email = new FormControl('', [Validators.required, Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$')]);
+  password = new FormControl('', [Validators.required]); 
+  hide = true;
```
Comment (MuskaaanBansal) [inline src/app/components/login-dialog/login-dialog.component.ts]:
```suggestion
  hidePassword = true;
```

**Review by MuskaaanBansal — CHANGES_REQUESTED** [review]
install prettier to lint all your files

**Pull request merged**